Mathlib/Data/Vector/Basic.lean

```diff
@@ -79,6 +79,16 @@ theorem mk_toList : ∀ (v : Vector α n) (h), (⟨toList v, h⟩ : Vector α n)
 --     ((coe : { l : List α // l.length = n } → List α) v).length = n :=
 --   v.2
 
+@[simp]
+theorem pmap_cons {p : α → Prop} (f : (a : α) → p a → β) (a : α) (v : Vector α n)
+    (hp : ∀ x ∈ (cons a v).toList, p x) :
+    (cons a v).pmap f hp = cons (f a (by
+      simp only [Nat.succ_eq_add_one, toList_cons, List.mem_cons, forall_eq_or_imp] at hp
+      exact hp.1))
+      (v.pmap f (by
+        simp only [Nat.succ_eq_add_one, toList_cons, List.mem_cons, forall_eq_or_imp] at hp
+        exact hp.2)) := rfl
```

Comment:
Can you write a version of this lemma backwards deriving `∀ x ∈ (cons a v).toList, p x` from `∀ x ∈ v.toList, p x` and `p a`, rather than the other way around?